content/common/utils/utils.h

```diff
@@ -0,0 +1,64 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+
+#ifndef UTILS_H_
+#define UTILS_H_ 1
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef __FUNCTION_NAME__
+#ifdef WIN32  /* WINDOWS */
+	#define __FUNCTION_NAME__   __FUNCTION__  
+#else  /* *NIX */
+	#define __FUNCTION_NAME__   __func__ 
+#endif
+#endif
+
+#if defined(__linux__) || defined(__Unikraft__)
+#define print_err(call_desc)	perror(call_desc)
+#elif defined(_WIN32)
+#include <Windows.h>
+
+static VOID PrintLastError(const PCHAR message)
+{
+	CHAR errBuff[1024];
+
+	FormatMessage(
+		FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_MAX_WIDTH_MASK,
+		NULL,
+		GetLastError(),
+		0,
+		errBuff,
+		sizeof(errBuff) - 1,
+		NULL);
+
+	fprintf(stderr, "%s: %s\n", message, errBuff);
+}
+
+#define print_err(call_desc)	PrintLastError(call_desc)
+#endif
+
+#define DIE(assertion, call_description)	\
+	do {					\
+		if (assertion) {		\
+			ERR(call_description);	\
+			exit(EXIT_FAILURE);	\
+		}				\
+	} while (0)
+
+#define ERR(call_description)					\
+	do {							\
+		fprintf(stderr, "(%s, %s, %d): ",		\
+			__FILE__, __FUNCTION_NAME__, __LINE__);	\
+		print_err(call_description);			\
```

Comment:
We would replace `print_err` with `get_last_error` that provides the string. And we would format the error message by using that string. We rely on `errno` / `strerror` for this, a generic / portable way of treating errors.